pyanaconda/modules/localization/localed.py

```diff
@@ -300,16 +246,129 @@ def convert_layouts(self, layouts_variants):
+
+            # layouts has changed in compositor, always emit this signal
+            log.debug("Localed layouts has changed. Last known: '%s' current: '%s'",
+                      self._last_layouts_variants, layouts_variants)
+            self.compositor_layouts_changed.emit(layouts_variants)
+
+            # check if last selected variant has changed
+            # nothing is selected in compositor
+            if not layouts_variants:
+                log.warning("Compositor layouts not set.")
+                self.compositor_selected_layout_changed.emit("")
+            # we don't know last used layouts
+            elif not self._last_layouts_variants:
+                log.debug("Compositor selected layout is different. "
+                          "Missing information about last selected layouts.")
+                self.compositor_selected_layout_changed.emit(layouts_variants[0])
+            # selected (first) has changed
+            elif layouts_variants[0] != self._last_layouts_variants[0]:
+                log.debug("Compositor selected layout is different.")
+                self.compositor_selected_layout_changed.emit(layouts_variants[0])
+
+            self._last_layouts_variants = layouts_variants
+
+    @property
+    def current_layout_variant(self):
+        """Get first (current) layout with variant.
+
+        :return: a list of "layout (variant)" or "layout" layout specifications
+        :rtype: list(str)
+        """
+        return "" if not self.layouts_variants else self.layouts_variants[0]
+
+    @property
+    def layouts_variants(self):
+        """Get current X11 layouts with variants.
+
+        Store information about the last selected layout variant used.
+
+        :return: a list of "layout (variant)" or "layout" layout specifications
+        :rtype: list(str)
+        """
+        self._last_layouts_variants = super().layouts_variants
+        return self._last_layouts_variants
+
+    def set_layouts(self, layouts_variants, options=None, convert=False):
+        """Set X11 layouts.
+
+        Store user selected layouts with variants.
+
+        :param layouts_variants: list of 'layout (variant)' or 'layout'
+                                 specifications of layouts and variants
+        :type layouts_variants: list(str)
+        :param options: list of X11 options that should be set
+        :type options: list(str)
+        :param convert: whether the layouts should be converted to a VConsole keymap
+                        (see set_and_convert_layouts)
+        :type convert: bool
+        """
+        self._set_layouts(layouts_variants, options, convert)
+        log.debug("Storing layouts for compositor configured by user")
+        self._user_layouts_variants = layouts_variants
+
+    def _set_layouts(self, layouts_variants, options=None, convert=False):
+        """Set a new layouts but these are not user defined.
```

Comment:
This doc string was a bit confusing to me.
I even think it is quite clear without it and the comment below - if I understand it correctly: layouts can be set either by calling set_layouts (user request to set layouts) or set_current_layout (user request to set current layout).

Reply:
I tried to improve it.